src/Cli.ts

```diff
@@ -0,0 +1,89 @@
+import yargs from "yargs";
+import fs from "fs";
+import {OutputFormat} from "./OutputFormat";
+
+/** common CLI arguments structure */
+export class IonCliCommonArgs {
+    inputFiles: Array<string>;
+    outputFile: any;
+    outputFormatName: OutputFormat;
+    errorReportFile: any;
+
+    constructor(argv: yargs.Arguments) {
+        this.outputFile = argv["output"] ? fs.createWriteStream(<string>argv["output"], {flags: 'w'}) : process.stdout;
+        // create error output stream (DEFAULT: stderr)
+        this.errorReportFile = argv["error-report"] ? fs.createWriteStream(<string>argv["error-report"], {flags: 'w'}) : process.stderr;
+        this.outputFormatName = <OutputFormat>argv["output-format"];
+        this.inputFiles = <Array<string>>argv["inputfile"];
+    }
+
+    getOutputFile(): any {
+        return this.outputFile;
+    }
+
+    getErrorReportFile(): any {
+        return this.errorReportFile;
+    }
+
+    getInputFiles(): Array<string> {
+        return this.inputFiles;
+    }
+
+    getOutputFormatName(): OutputFormat {
+        return this.outputFormatName;
+    }
+}
+
+/** Error Types symbol[READ | WRITE | STATE] */
+export enum ErrorType {
+    READ = "READ",
+    WRITE = "WRITE",
+    STATE = "STATE"
+}
+
+/** Error structure for error report */
+export class IonCliError {
+    errorType: ErrorType;
+    location: string;
+    message: string;
+    errorReportFile: any;
+
+    constructor(errorType: ErrorType, location: string, message: string, errorReportFile: any) {
+        this.errorType = errorType;
+        this.location = location;
+        this.message = message;
+        this.errorReportFile = errorReportFile;
+    }
+
+    writeErrorReport() {
+        this.errorReportFile.write(this.errorType + ": " + this.location + " " + this.message + "\n");
+    }
+}
+
+const argv = yargs
+    .commandDir(__dirname,{
+        extensions: ['ts'],
+    })
+    .options({
```

Comment:
How do you differentiate between the process command and other commands?

Reply:
To differentiate between commands I have used a constant called command and a handler for the command file like Process.ts(8-16) which has a handler stating to call process.(Yargs support having a command directory in which you can have command files specified using a command and a handler constants).